pallets/xyk/runtime-api/src/lib.rs

```diff
@@ -106,5 +116,7 @@ sp_api::decl_runtime_apis! {
 			path: sp_std::vec::Vec<TokenId>,
 			input_amount: Balance,
 		) -> Option<bool>;
+
+		fn get_tradeable_tokens() -> Vec<(TokenId, RpcAssetMetadata)>;
```

Comment:
as you already introduced new type consider putting TokenId into RpcAssetMetadata as well

Reply:
Refactored.